### ISSUE TITLE
Bash commands entered while previous command is executing doesn't resolve afterwards (2.42)

### ISSUE THREAD

**Issue by ckorris-imt** [git-for-windows/git#4603]
While using the Bash terminal in 2.42, I am not able to "stack" commands, where if I enter a command that takes some time to execute, and while it's executing, enter a second command and press enter, the second command will resolve after the first is done. In 2.42, the second command is ignored. 

This happened after upgrading after a fresh Windows install. However, I verified that going back to 2.41 removes the issue, and then coming back to 2.42 brings the issue back. I use all default settings. 

 - [ ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing. 

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

Version 2.42.0. Uninstalling to 2.41.0 solved the problem. I had previously tried uninstalling and reinstalling, clearing my settings, etc. Doing the exact same thing but with 2.41.0 fixed it, Doing the exact same thing again with 2.42 brought it back. 

```
$ git --version --build-options

git version 2.42.0.windows.2
cpu: x86_64
built from commit: 2f819d1670fff9a1818f63b6722e9959405378e3
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon


```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 10 64-bit

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19045.3324]
(c) Microsoft Corporation. All rights reserved.
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

I chose all the defaults - MinTTY, Vim, etc. 

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Possibly relevant, note that I uninstalled 2.41 and reinstalled 2.42 for the purposes of verifying that it was the problem and also so that I could fill out this bug report, hence not much being configured. But I can indeed reproduce the issue again. 

$ "C:\Program Files\Git\etc\install-options.txt"
C:\Program Files\Git\etc\install-options.txt: line 1: Editor: command not found
C:\Program Files\Git\etc\install-options.txt: line 2: Custom: command not found
C:\Program Files\Git\etc\install-options.txt: line 3: Default: command not found
C:\Program Files\Git\etc\install-options.txt: line 4: Path: command not found
SSH: Could not resolve hostname Option:: Name or service not known
C:\Program Files\Git\etc\install-options.txt: line 6: Tortoise: command not found
curl: (3) URL rejected: Port number was not a decimal number between 0 and 65535
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0curl: (6) Could not resolve host: OpenSSL
C:\Program Files\Git\etc\install-options.txt: line 8: CRLF: command not found
Bash: Terminal: No such file or directory
git: 'Pull' is not a git command. See 'git --help'.

The most similar command is
        pull
C:\Program Files\Git\etc\install-options.txt: line 11: Use: command not found
C:\Program Files\Git\etc\install-options.txt: line 12: Performance: command not found
C:\Program Files\Git\etc\install-options.txt: line 13: Enable: command not found
C:\Program Files\Git\etc\install-options.txt: line 14: Enable: command not found
C:\Program Files\Git\etc\install-options.txt: line 15: Enable: command not found

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash. I have not tried others.

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
It happens with any combination of commands where the first takes enough time for you to be able to enter a second command and press enter. The combo I use the most to test is `git fetch` followed by `git status`. But it happened with plenty of other combinations, I have not seen it work as expected with any combination. 
```
 - What did you expect to occur after running these commands?

After typing `git fetch`, pressing enter, and then typing `git status` and pressing enter _before_ the fetch was finished, I expected it to then complete the status command afterwards.

 - What actually happened instead?

After the fetch was done, it did not do anything, as if I had not entered a second command. 

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

It it not repository specific.

**Comment by dscho** [comment]
If this reproduces with regular [Cygwin](https://cygwin.com), it is probably something we have to live with.

**Comment by dscho** [comment]
> If this reproduces with regular [Cygwin](https://cygwin.com), it is probably something we have to live with.

@ckorris-imt does it reproduce with regular Cygwin?

**Comment by emerick** [comment]
This has been happening to me as well. I started noticing it in 2.41. I feel like it got slightly better in 2.42, but it's still a major problem. I'm unable to repro the issue in Cygwin 3.4.9.

**Comment by dscho** [comment]
> This has been happening to me as well. I started noticing it in 2.41. I feel like it got slightly better in 2.42, but it's still a major problem. I'm unable to repro the issue in Cygwin 3.4.9.

Did you enable pseudo Console support?

**Comment by emerick** [comment]
> Did you enable pseudo Console support?

Not that I'm aware of. I just installed the same as always. This is on Windows 11, BTW.

**Comment by dscho** [comment]
> > Did you enable pseudo Console support?
> 
> Not that I'm aware of. I just installed the same as always. This is on Windows 11, BTW.

For various reasons, Git for Windows still defaults to turning off Pseudo Console support, while Cygwin is "all in". Could you reinstall Git for Windows, toggling Pseudo Console support on, and test again, please?

**Comment by ckorris-imt** [comment]
That is confirmed to be the fix on my end, as well. Thank you!

**Comment by dscho** [comment]
I guess we'll have to switch to enabling Pseudo Console support for all ☹️

**Comment by kriek** [comment]
> I guess we'll have to switch to enabling Pseudo Console support for all ☹️

What are the Pseudo Console activation drawbacks?

**Comment by dscho** [comment]
> What are the Pseudo Console activation drawbacks?

I am still not convinced of the quality of the code. There has been a relatively steady stream of bug reports followed up by fixes that were neither well-described nor well-contained, it often seemed as if they were only addressing symptoms and not root causes.

But that steady stream _did_ wane a little. What has increased are the problems with non-Pseudo Console code that was supposed to be left alone by Pseudo Console support but wasn't.

So at this stage it might be a "damned if you don't, damned if you do" situation.

**Comment by mfriedrich74** [comment]
For how long was the pseudo console support enabled due to another bug that
switched the setting? Maybe factor that in in your decision? It may have
been longer under test than thought.

Best regards, Mike

**Comment by dscho** [comment]
> For how long was the pseudo console support enabled

As per the explanation in https://github.com/git-for-windows/build-extra/pull/522, it was enabled by mistake in v2.41.0 and that bug was fixed in v2.42.0(2). That's equivalent to about 8 weeks and while v2.41.0's installer was downloaded over 2.7 million times, the v2.42.0 one was downloaded only just over 800k times. And since I cannot add any telemetry without having to fear for my own life at the hands of some very opinionated people, I cannot say how many of those downloaded installers were actually run, let alone in how many instances they were followed by frustrated downgrades to the previous Git for Windows version.

In other words: I am deprived of any dependable data to support making that decision.

**Comment by GeorgievGG** [comment]
Does "Pseudo Console" activation resolve this problem for higher versions than 2.42?
